modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/GridCacheOffheapManager.java

```diff
@@ -1108,7 +1110,30 @@ private Metas getOrAllocateCacheMetas() throws IgniteCheckedException {
         try {
             int cleared = 0;
 
-            for (CacheDataStore store : cacheDataStores()) {
+            Iterable<CacheDataStore> cacheDataStores = cacheDataStores();
+
+            // Use random shift to reduce contention.
+            int shift = ThreadLocalRandom.current().nextInt(F.size(cacheDataStores.iterator()));
+
+            int cnt = 0;
+            for (CacheDataStore store : cacheDataStores) {
```

Comment:
This doesn't seem to work. After executing F.size() cacheDataStores doesn't return elements.

Reply:
Thanks, fixed